### PR TITLE
add doc for template preprocessor extension

### PR DESCRIPTION
**Pull request by superyyrrzz**
add doc for https://github.com/dotnet/docfx/pull/811
@chenkennt @vwxyzh @hellosnow @qinezh @ansyral

### CODE REVIEW

File: src/docfx.website.themes/default/conceptual.extension.js [hunk 1]
```diff
@@ -1,9 +1,15 @@
 // Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+/**
+ * This method will be called at the start of exports.transform in conceptual.html.primary.js
+ */
 exports.preSteps = function (model) {
```
Comment (vwxyzh) [inline src/docfx.website.themes/default/conceptual.extension.js]: >preSteps [](start = 8, length = 8)

pretransform?

File: Documentation/tutorial/howto_create_custom_template.md [hunk 2]
```diff
@@ -90,3 +90,25 @@ Open http://localhost:8080/toc.html and you can see a toc web page.
 > Tip: Run `docfx template export default` to view what's inside the default template.
 
 > NOTE: It is possible that DocFX updates its embedded templates when a new version is released. So please make sure to re-export the template if you overwrite or are dependent on it in your custom template.
+
+Extension for *Preprocessor* file
+----------------------------------
+If you want to modify some properties based on DocFX `default` template's *Preprocessor*, you can use *Preprocessor* extension file to achieve this.
+For example, if you want to add a property to the managed reference's data model after `default` template's *Preprocessor*, you can update the file `ManagedReference.extension.js` in your customized template with the following content:
```
Comment (qinezh) [inline Documentation/tutorial/howto_create_custom_template.md]: Add an empty line here if you want split it into two lines

**Comment by ansyral** [comment]
:shipit:

File: Documentation/tutorial/howto_create_custom_template.md [hunk 3]
```diff
@@ -90,3 +90,26 @@ Open http://localhost:8080/toc.html and you can see a toc web page.
...
+
+/**
+ * This method will be called at the end of exports.transform in ManagedReference.html.primary.js
+ */
+exports.postTransform = function (model) {
+  model._extra_property = "Hello world";
+  return model;
+}
+```
+Compared with modifying `ManagedReference.html.primary.js` directely, you needn't worry about merging your customized template with DocFX's embedded templates when DocFX updates.
```
Comment (hellosnow) [inline Documentation/tutorial/howto_create_custom_template.md]: >template [](start = 118, length = 8)

templates?

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**